velox/exec/PartitionedOutputBufferManager.cpp

```diff
@@ -532,8 +532,8 @@ void PartitionedOutputBufferManager::deleteResults(
         }
         return it->second;
       });
-  if (buffer && buffer->deleteResults(destination)) {
-    buffers_.withLock([&](auto& buffers) { buffers.erase(taskId); });
+  if (buffer) {
+    buffer->deleteResults(destination);
```

Comment:
Does this mean that the same TaskId can be deleted multiple times? Can we look into why this could happen by logging?

Reply:
Can you clarify "same TaskId can be deleted multiple times"? 
FYI, this part of the code gets called once for each destination that is getting aborted (by the downstream tasks when they pre-maturely finish). It's not really trying to delete the task, but just mark the buffers as being finished.

I'm guessing the assumption in the original code is that we want to remove the buffer as soon as possible (e.g. before the driver/task finishes). However, the async nature of this abort call from the downstream causes issues in the current driver/task. Through testing, I've found that we really don't need to remove the buffer and we can allow the natural progression of the task to actually delete the buffer for the task.

FYI, I've added some logs to confirm this (I put it at the top of https://github.com/facebookincubator/velox/issues/3009).

Reply:
> Does this mean that the same TaskId can be deleted multiple times, say once during abortResult
I don't think the task is being "deleted" currently... (nor does it need to). The current code was trying to clean the buffers up too soon. The actual deletion (e.g. clean up of resources for the task) should happen in task::Terminate.